### PR TITLE
Support AOT mode in GTK shell

### PR DESCRIPTION
**Pull request by robert-ancell**
As copied from #18423

### CODE REVIEW

**Comment by robert-ancell** [comment]
Note this breaks the existing API for `dart_project_new()` - please check if the new API makes more sense.

File: shell/platform/linux/public/flutter_linux/fl_dart_project.h [hunk 1]
```diff
@@ -24,44 +24,56 @@ G_DECLARE_FINAL_TYPE(FlDartProject, fl_dart_project, FL, DART_PROJECT, GObject)
...
+ * @library_path: (type filename): a directory name, e.g. "lib".
  *
- * Creates a Flutter project. The project path should contain the following
+ * Creates a Flutter project. The data path should contain the following
  * top-level items:
  * - icudtl.dat (provided as a resource by the Flutter tool)
- * - flutter_assets (as built by the Flutter tool)
+ * - flutter_assets/ (as built by the Flutter tool)
+ * The library path should contain:
+ * - libapp.so (if AOT compiled mode in use).
```
Comment (stuartmorgan) [inline shell/platform/linux/public/flutter_linux/fl_dart_project.h:35]: Is there a reason this isn't using the same approach that the C++ version uses, where the user provides a single root directory, and the locations of each piece inside that are pre-defined? That simplifies the runner.

Comment (robert-ancell) [reply shell/platform/linux/public/flutter_linux/fl_dart_project.h]: Based on discussion on Discord I've removed all the path arguments and set them to the expected standard bundle layout.

File: shell/platform/linux/public/flutter_linux/fl_dart_project.h [hunk 2]
```diff
@@ -24,44 +24,23 @@ G_DECLARE_FINAL_TYPE(FlDartProject, fl_dart_project, FL, DART_PROJECT, GObject)
 
 /**
  * fl_dart_project_new:
- * @path: a file path, e.g. "my_dart_project".
  *
- * Creates a Flutter project. The project path should contain the following
- * top-level items:
- * - icudtl.dat (provided as a resource by the Flutter tool)
- * - flutter_assets (as built by the Flutter tool)
```
Comment (stuartmorgan) [inline shell/platform/linux/public/flutter_linux/fl_dart_project.h:32]: Documenting what it expects (the files, the fact that it's executable-relative) is still useful.

**Pull request merged**